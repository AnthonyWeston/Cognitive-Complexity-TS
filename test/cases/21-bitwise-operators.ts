function allAnd() {
    1 & 1 & 1;
}

function allOr() {
    1 | 1 | 1;
}

function twoSequences() {
    1 & 1 & 1 | 1 | 1;
}

function threeSequences() {
    1 & 1 | 1 | 1 & 1;
}

function noOp() {
    1;
}
