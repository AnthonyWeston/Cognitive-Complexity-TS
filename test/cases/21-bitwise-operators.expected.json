{
    "21-bitwise-operators.ts": {
        "score": 7,
        "inner": [
            {
                "name": "allAnd",
                "score": 1,
                "line": 1,
                "column": 1
            },
            {
                "name": "allOr",
                "score": 1,
                "line": 5,
                "column": 1
            },
            {
                "name": "twoSequences",
                "score": 2,
                "line": 9,
                "column": 1
            },
            {
                "name": "threeSequences",
                "score": 3,
                "line": 13,
                "column": 1
            },
            {
                "name": "noOp",
                "score": 0,
                "line": 17,
                "column": 1
            }
        ]
    }
}
